src/query/api/v1/handler/database/kvstore.go

```diff
@@ -199,7 +199,7 @@ func (h *KeyValueStoreHandler) update(
 func (h *KeyValueStoreHandler) newKVProtoMessage(key string) (protoiface.MessageV1, error) {
 	if h.kvStoreProtoParser != nil {
 		v, err := h.kvStoreProtoParser(key)
-		if err != nil {
+		if err == nil {
 			return v, nil
```

Comment:
hmm do we want to return an error if we do encounter one though?

Reply:
Well the error just is indicating that the key doesn't exist so we fallback to the based protos